File: code/java/剑指offer10-1.java
class Solution {
    public int fib(int n) {
        if(n<=1) return n;
        if(n<=2) return 1;

        long a = 0;
        long b = 1;
        for(long i=2;i<=n;i++){
            long temp = b;
            b = (b+a)%((long)1e9+7);
            a = temp;
        }
        return (int)b;
    }
}

File: code/python/剑指offer10-1.py
class Solution:
    def fib(self, n: int) -> int:
        if n<=1:
            return n
        if n==2:
            return 1
        a = 0
        b = 1
        for i in range(2,n+1):
            temp = b
            b = (b+a)%(1e9+7)
            a = temp
        #print(a,b)
        return int(b)